smart_open/tests/test_s3_version.py

```diff
@@ -70,8 +70,10 @@ def test_good_id(self):
     def test_bad_id(self):
         """Does passing an invalid version_id exception into the s3 submodule get handled correctly?"""
         params = {'version_id': 'bad-version-does-not-exist'}
-        with self.assertRaises(IOError):
-            open(self.url, 'rb', transport_params=params)
+        with open(self.url, 'rb', transport_params=params) as fin:
+            with self.assertRaises(IOError):
```

Comment:
We should not change this test. Opening the URL to a non-existent object should raise an error, before any reading is attempted.

Reply:
So I hadn't changed this in my fork of the code. But then again I wasn't aiming to make a PR. I could check what errors the fix raise during tests and see if there is a sensible way to incorporate them without changing core functionality.